### PR TITLE
Fix writing dark colors

### LINKED ISSUE #3821: Dark colors are printed incorrectly with FormattedTextOutputDeviceData
I noticed that dark colors (that is, `Black` and all the color names starting with `Dark`) are printed incorrectly, and are rendered as their lighter counterparts instead of the dark versions.

Here is a comparison between `Console.WriteLine` and `IOutputDevice.DisplayAsync`:

image

The `Console` version:

```csharp
Console.WriteLine("This is the default color");

foreach (var consoleColor in Enum.GetValues<ConsoleColor>())
{
    Console.ForegroundColor = consoleColor;
    Console.WriteLine("This is foreground {0}", consoleColor);
}
```

The `IOutputDevice` version:

```csharp
await outputDevice.DisplayAsync(this, new TextOutputDeviceData("This is the default color"));

foreach (var consoleColor in Enum.GetValues<ConsoleColor>())
    await outputDevice.DisplayAsync(
        this,
        new FormattedTextOutputDeviceData(
            "This is foreground " + consoleColor.ToString()
        ) { ForegroundColor = new SystemConsoleColor { ConsoleColor = consoleColor } }
    );
```

I've attached the repro project.

[testfx.zip](https://github.com/user-attachments/files/16985841/testfx.zip)

### PR DESCRIPTION
**Pull request by nohwnd**
Fix writing dark colors as their bright counterpart. 
Fix writing both grays as dark gray.

Fix #3821

### CODE REVIEW

File: test/UnitTests/Microsoft.Testing.Extensions.UnitTests/Properties/launchSettings.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "profiles": {
     "Microsoft.Testing.Extensions.UnitTests": {
       "commandName": "Project",
-      "commandLineArgs": "--treenode-filter /*/*/*/**"
+      "commandLineArgs": "--treenode-filter /*/*/*/** --output detailed"
```
Comment (MarcoRossignoli) [inline test/UnitTests/Microsoft.Testing.Extensions.UnitTests/Properties/launchSettings.json]: do you want to merge this one too?

**Review by MarcoRossignoli — APPROVED** [review]

File: test/UnitTests/Microsoft.Testing.Extensions.UnitTests/Properties/launchSettings.json (hunk 1 above)
Comment (nohwnd) [reply test/UnitTests/Microsoft.Testing.Extensions.UnitTests/Properties/launchSettings.json]:
```suggestion
      "commandLineArgs": "--treenode-filter /*/*/*/**
```

Comment (Evangelink) [reply test/UnitTests/Microsoft.Testing.Extensions.UnitTests/Properties/launchSettings.json]: Let's do it on all test files as a separate PR. I think that's a good default.

File: test/UnitTests/Microsoft.Testing.Extensions.UnitTests/Properties/launchSettings.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "profiles": {
     "Microsoft.Testing.Extensions.UnitTests": {
       "commandName": "Project",
-      "commandLineArgs": "--treenode-filter /*/*/*/**"
+      "commandLineArgs": "--treenode-filter /*/*/*/**
```
Comment (nohwnd) [inline test/UnitTests/Microsoft.Testing.Extensions.UnitTests/Properties/launchSettings.json]:
```suggestion
      "commandLineArgs": "--treenode-filter /*/*/*/**"
```

**Review by MarcoRossignoli — APPROVED** [review]

**Review by Evangelink — APPROVED** [review]
LGTM! I think we should invest in some kind of screenshot diff acceptance tests to avoid color regression (later).

**Comment by nohwnd** [comment]
@evangelink, backport to previous version?

**Comment by Evangelink** [comment]
> @Evangelink, backport to previous version?

Yeah you will probably need to release `1.4.1` so backport to `rel/1.4` (don't force merge yet, I need to bump versions)

**Pull request merged**